### PR TITLE
Adding public final ServiceDiscoveryManager instance to MultiUserChatManager

### PR DESCRIPTION
**Pull request by adiaholic**
The expensive method ServiceDiscoveryManager.getInstanceFor() is replaced by a final private field

### CODE REVIEW

File: smack-extensions/src/main/java/org/jivesoftware/smackx/muc/MultiUserChatManager.java [hunk 1]
```diff
@@ -82,7 +82,7 @@
     private static final String DISCO_NODE = MUCInitialPresence.NAMESPACE + "#rooms";
 
     private static final Logger LOGGER = Logger.getLogger(MultiUserChatManager.class.getName());
-
+    private final ServiceDiscoveryManager serviceDiscoveryManager;
```
Comment (Flowdalic) [inline smack-extensions/src/main/java/org/jivesoftware/smackx/muc/MultiUserChatManager.java]: The declaration should be not with the block auf static field declarations, but further down where the non-static fields are declared (e.g. just above the first constructor). Also there should be a newline before the next non-empty line comes.

**Comment by vanitasvitae** [comment]
Plus the Jive issue number ;)

**Comment by Flowdalic** [comment]
> Plus the Jive issue number ;)

Ahh good point. Assuming you mean Jira, not Jive. But anyway, please mentioned the Smack issue number that is related to the commit somewhere in the commit body so that Jira automatically links the commit.

**Comment by Flowdalic** [comment]
Squash into a single commit please. I am also not happy about the commit title. It really shouldn't continue in the commit message body if possible.

**Pull request closed without merging**